### ISSUE TITLE
implement performance metrics collection

### ISSUE THREAD

**Issue by jwcbmat** [jwcbmat/efficiency-node#1]
Develop functionality to collect performance metrics such as response time, CPU and memory usage, and network traffic for APIs using Node.js.

**Tasks:**
- Define key metrics for monitoring.
- Implement response time collection logic using `Performance measurement APIs`.
- Use Node.js modules (`os` and `process`) to monitor CPU and memory usage.
- Integrate metrics collection with an HTTP server for testing.
- Create temporary storage for collected metrics.
- Write unit tests to validate metrics collection.

**Comment by jwcbmat** [comment]
Initial key metrics:

    - Response Time: The time it takes for the API to respond to a request.
    - CPU Usage: Percentage of CPU usage by the Node.js process.
    - Memory Usage: Amount of memory used by the Node.js process.
    - Network Traffic: Volume of data sent and received by the API.

**Comment by jwcbmat** [comment]
By tracking response time, CPU usage, memory usage, and network traffic, we can effectively measure API performance under various loads and identify any potential bottlenecks or issues that need attention. These metrics will help us ensure that our API runs efficiently and reliably for our users.

**Comment by jwcbmat** [comment]
simple middleware ready

    - Metrics Collection: Uses the MetricsCollector class to gather data like CPU usage, memory usage, and response time.
    - Conditional Logging: If logMetrics is true, the collected metrics are printed to the console.
    - Single Execution Control: Ensures that metrics collection and logging occur only once per request.

**Comment by jwcbmat** [comment]
> Develop functionality to collect performance metrics such as response time, CPU and memory usage, and network traffic for APIs using Node.js.
> 
> **Tasks:**
> 
>     * Define key metrics for monitoring.
> 
>     * Implement response time collection logic using `Performance measurement APIs`.
> 
>     * Use Node.js modules (`os` and `process`) to monitor CPU and memory usage.
> 
>     * Integrate metrics collection with an HTTP server for testing.
> 
>     * Write unit tests to validate metrics collection.

Update ´* Create temporary storage for collected metrics´:

Instead, I'll open a new issue to address this aspect separately.
